client/my-sites/media-library/upload-url.jsx

```diff
@@ -13,71 +14,70 @@ import analytics from 'lib/analytics';
 import FormTextInput from 'components/forms/form-text-input';
 import MediaActions from 'lib/media/actions';
 
-export default React.createClass( {
-	displayName: 'MediaLibraryUploadUrl',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		onAddMedia: React.PropTypes.func,
-		onClose: React.PropTypes.func,
-		className: React.PropTypes.string
-	},
-
-	getInitialState() {
-		return {
-			value: '',
-			isError: false
-		};
-	},
-
-	getDefaultProps() {
-		return {
-			onAddMedia: noop,
-			onClose: noop
-		};
-	},
-
-	upload( event ) {
-		var isError = ! event.target.checkValidity();
+class MediaLibraryUploadUrl extends Component {
+	static propTypes = {
+		className: PropTypes.string,
+		site: PropTypes.object,
+		onAddMedia: PropTypes.func,
+		onClose: PropTypes.func,
+	};
 
-		this.setState( {
-			isError: isError
-		} );
+	static defaultProps = {
+		onAddMedia: noop,
+		onClose: noop,
+	};
+
+	state = {
+		value: '',
+		isError: false,
+	};
+
+	upload = event => {
+		const isError = ! event.target.checkValidity();
+
+		this.setState( { isError } );
 
 		if ( isError || ! this.props.site ) {
+			event.preventDefault();
```

Comment:
Just a quick drive-by comment - it seems we need to call `event.preventDefault()` in both cases, so it should probably be called once (before this if branch) - otherwise someone might think that one of these calls should not be there :)